djangoplicity/pages/tests/test_models.py

```diff
@@ -0,0 +1,150 @@
+from django.conf import settings
+from datetime import datetime, timedelta
+from django.test import TestCase
+from factories import SectionFactory, PageFactory, URLFactory, PageGroupFactory, EmbeddedPageKeyFactory
+from djangoplicity.pages.models import (
+    Page, URL, EmbeddedPageKey, Section,
+    register_page_key, generate_page_id, page_cache_invalidator, 
+    url_cache_invalidator, page_key_cache_invalidator, section_cache_invalidator
+    )
+
+from mock import patch
+
+class SectionTest(TestCase):
+
+    def test_unicode_representation(self):
+        section = SectionFactory.build()
+        self.assertEqual(u'%s' % section, 'TestingSection')
+
+
+class PageTest(TestCase):
+
+    def test_unicode_representation(self):
+        page = PageFactory.build()
+        self.assertEqual(u'%s' % page, 'TestingTitle')
+
+    def test_is_online(self):
+        page = PageFactory.build()
+
+        page.published = False
+        self.assertFalse(page.is_online())
+
+        page.published = True
+        self.assertTrue(page.is_online())
+
+        page.start_publishing = datetime.now() + timedelta(days=1)
+        self.assertFalse(page.is_online())
+        page.start_publishing = None
+
+        page.end_publishing = datetime.now() - timedelta(days=1)
+        self.assertFalse(page.is_online())
+        page.end_publishing = None
+
+        # translatedPage = PageFactory.create()
+        # translatedPage.source = PageFactory.create()
+        
+        # self.assertTrue(translatedPage.is_online())
+
+    def test_get_absolute_url(self):
+        translatedPage = PageFactory.create()
+        translatedPage.source = PageFactory.create()
+        self.assertIsNotNone(translatedPage.get_absolute_url())
+
+        translatedPage.source = None
+        self.assertIsNotNone(translatedPage.get_absolute_url())
+
+
+    def test_save(self):
+        section = SectionFactory.build()
+        section.save()
+        page = PageFactory.build()
+        page.script = '<script></script>'
+        page.section = section
+        page.save()
+
+        self.assertFalse(page.live_edit)
+
+    def test_test_render_errors(self):
+        page = PageFactory.build()
+        page.test_render_errors()
+
+
+class URLTest(TestCase):
+    def test_unicode_representation(self):
+        url = URLFactory.build()
+        self.assertEqual(u'%s' % url, 'http://someurl.com') 
+
+class PageGroupTest(TestCase):
+    def test_unicode_representation(self):
+        page_group = PageGroupFactory.build()
+        self.assertEqual(u'%s' % page_group, 'TestingName')
+
+class EmbeddedPageKeyTest(TestCase):
+    def test_unicode_representation(self):
+        page_key = EmbeddedPageKeyFactory.build()
+        self.assertEqual(
+            u'%s' % page_key, u'%s (%s)' % (page_key.title, page_key.page_key))
```

Comment:
u strings and redundant and troublesome in Python 3, please don't use it and prefer the format method instead
```suggestion
            page_key, '{} ({})'.format(page_key.title, page_key.page_key))
```

Reply:
Resolved in [#ad182f52](https://github.com/djangoplicity/djangoplicity/pull/96/commits/ad182f52f0a9dbcfe1655e63e109986108265504)